### PR TITLE
go.mod: update to k8s 1.27.1 and controller runtime 0.15.x 

### PR DESCRIPTION
**Pull request by mythi**
Fixes: #1249 

k8s 1.27.x triggers build errors on controller-runtime 0.14.x
so we will need to update to 0.15.x at the same time.

Changes include:

* k8s e2e framework moved to use Ginkgo context so we add
  test context to all our test nodes.
* adapt Ginkgo parameter modifications.
* adapt SGX admissionwebhook to InjectDecoder removal.
* adapt deviceplugins and FPGA CRDs to controller-runtime
  API changes.

### CODE REVIEW

**Comment by eero-t** [comment]
Go 1.20 changes look straightforward, but "k8s 1.27.0-alpha.1" changes seem to have triggered a lot of errors (which could be documented better).  Is this PR still valid?

**Comment by mythi** [comment]
> Go 1.20 changes look straightforward, but "k8s 1.27.0-alpha.1" changes seem to have triggered a lot of errors (which could be documented better). Is this PR still valid?

It's a draft. Blocked by controller-runtime's client-go changes to get things unblocked.

**Comment by mythi** [comment]
@tkatila @hj-johannes-lee ready for review, PTAL

File: deployments/operator/webhook/manifests.yaml
Comment (tkatila) [inline deployments/operator/webhook/manifests.yaml:174]: As discussed offline, this is seems funny but it fixing it doesn't seem to be an option (breaking change in the controller runtime).

**Review by hj-johannes-lee — APPROVED** [review]

**Comment by hj-johannes-lee** [comment]
the failing test is not related to this pr, right?

File: deployments/operator/webhook/manifests.yaml
Comment (mythi) [reply deployments/operator/webhook/manifests.yaml:174]: this is the way to go based on the CR maintainers so closing this review comment

**Pull request merged**
